### PR TITLE
Make flatten() more robust against crashes (fixes #83)

### LINKED ISSUE #83: flatten() raises a ValueError
We received the following bug report via e-mail:

===
Hello Kepler Science Time,

I am going through the 'Science with lightkurve' tutorial 'How to perform aperture photometry with custom apertures'. (http://lightkurve.keplerscience.org/tutorials/2.05-making-custom-apertures.html)

When I run the code in the tutorial in my notebook I am getting an error.

```
#Build the light curve
pipeline_lc = tpfs[0].to_lightcurve().flatten()
for tpf in tpfs:
    pipeline_lc = pipeline_lc.append(tpf.to_lightcurve().flatten())

#Clean the light curve
pipeline_lc = pipeline_lc.remove_nans().remove_outliers()

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-4ea5577ff332> in <module>()
      2 pipeline_lc = tpfs[0].to_lightcurve().flatten()
      3 for tpf in tpfs:
----> 4     pipeline_lc = pipeline_lc.append(tpf.to_lightcurve().flatten())
      5 
      6 #Clean the light curve

/Users/wesleysandidge/anaconda2/lib/python2.7/site-packages/lightkurve/lightcurve.pyc in flatten(self, window_length, polyorder, return_trend, break_tolerance, **kwargs)
    235             trend_signal[l:h] = signal.savgol_filter(x=lc_clean.flux[l:h],
    236                                                      window_length=window_length,
--> 237                                                      polyorder=polyorder, **kwargs)
    238         trend_signal = np.interp(self.time, lc_clean.time, trend_signal)
    239         flatten_lc = copy.deepcopy(self)

/Users/wesleysandidge/anaconda2/lib/python2.7/site-packages/scipy/signal/_savitzky_golay.pyc in savgol_filter(x, window_length, polyorder, deriv, delta, axis, mode, cval)
    335     if mode == "interp":
    336         if window_length > x.size:
--> 337             raise ValueError("If mode is 'interp', window_length must be less "
    338                              "than or equal to the size of x.")
    339 

ValueError: If mode is 'interp', window_length must be less than or equal to the size of x.
```
Is this a known error? What should I do to fix it?

### PR DESCRIPTION
**Pull request by barentsen**
This PR fixes #83!

### CODE REVIEW

File: lightkurve/lightcurve.py [hunk 1]
```diff
@@ -232,13 +232,22 @@ def flatten(self, window_length=101, polyorder=2, return_trend=False,
...
-                                                     window_length=window_length,
-                                                     polyorder=polyorder, **kwargs)
+                                                     window_length=segment_window_length,
+                                                     polyorder=segment_polyorder, **kwargs)
         trend_signal = np.interp(self.time, lc_clean.time, trend_signal)
         flatten_lc = copy.deepcopy(self)
-        flatten_lc.flux /= trend_signal
-        flatten_lc.flux_err /= trend_signal
+        flatten_lc.flux = flatten_lc.flux / trend_signal
+        flatten_lc.flux_err = flatten_lc.flux_err / trend_signal
```
Comment (mirca) [inline lightkurve/lightcurve.py]: Yo, I don't get the difference here xD

**Comment by mirca** [comment]
This look good to me! But I would raise that we should be careful not to turn a simple operation into a complex black-box.

File: lightkurve/lightcurve.py (hunk 1 above)
Comment (barentsen) [reply lightkurve/lightcurve.py]: The in-place operations are not the same; if you do `a += 1` it casts the result to fit back into a.  This was raising an exception when the types of `flux` and `trend_signal` were different.  For example, try this:

```
In [21]: a, b = np.array([1]), np.array([2.5])

In [22]: a /= b
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-22-b9f4626d3ca8> in <module>()
----> 1 a /= b

TypeError: ufunc 'true_divide' output (typecode 'd') could not be coerced to provided output parameter (typecode 'l') according to the casting rule ''same_kind''
```

**Comment by barentsen** [comment]
I totally agree it is scary that `flatten` now requires 30 lines of code.  We should try to keep operations as simple as possible.  This extra code is the price we pay for allowing flatten to work in segments, which, judging from the evidence @christinahedges posted in #62 does seem to be worth it?

File: lightkurve/tests/test_lightcurve.py [hunk 2]
```diff
@@ -360,3 +360,19 @@ def test_flatten():
     assert(len(flat_lc.time) == 5)
     assert(np.isfinite(flat_lc.flux).sum() == 3)
     assert(np.isfinite(flat_lc.flux_err).sum() == 3)
+
+
+def test_flatten_robustness():
+    """Test various special cases for flatten()."""
+    # flatten should work with integer fluxes
+    lc = LightCurve([1, 2, 3, 4, 5, 6], [10, 20, 30, 40, 50, 60])
+    flat_lc = lc.flatten(window_length=3, polyorder=1)
```
Comment (barentsen) [inline lightkurve/tests/test_lightcurve.py]: This line raises an exception if we use `flatten_lc.flux /= trend_signal` in the implementation as opposed to `flatten_lc.flux = flatten_lc.flux / trend_signal`.

File: lightkurve/lightcurve.py (hunk 1 above)
Comment (mirca) [reply lightkurve/lightcurve.py]: Oohhh, makes sense, thanks!

**Pull request merged**